src/s3/commands/uploadFile.ts

```diff
@@ -114,11 +193,167 @@ async function uploadWithProgress({
     )
 }
 
-function statFile(file: vscode.Uri) {
-    return statSync(file.fsPath).size
+interface BucketQuickPickItem extends vscode.QuickPickItem {
+    bucket: S3.Bucket | undefined
 }
 
-async function refreshNode(node: S3BucketNode | S3FolderNode, commands: Commands): Promise<void> {
-    node.clearChildren()
-    return commands.execute('aws.refreshAwsExplorerNode', node)
+// TODO:: extract and reuse logic from sam deploy wizard (bucket selection)
+/**
+ * Will display a quick pick with the list of all buckets owned by the user.
+ * @param s3client client to get the list of buckets
+ *
+ * @returns Bucket selected by the user, 'back' or 'cancel'
+ *
+ * @throws Error if there is an error calling s3
+ */
+export async function promptUserForBucket(
+    s3client: S3Client,
+    window = Window.vscode(),
+    promptUserFunction = promptUser,
+    createBucket = createBucketCommand
+): Promise<S3.Bucket | string> {
+    let allBuckets: S3.Bucket[]
+    try {
+        allBuckets = await s3client.listAllBuckets()
+    } catch (e) {
+        getLogger().error('Failed to list buckets from client', e)
+        window
+            .showErrorMessage(
+                localize(
+                    'AWS.message.error.promptUserForBucket.listBuckets',
+                    'Failed to list buckets from client, please try changing your region then try again'
+                ),
+                localize('AWS.message.prompt.changeS3RegionButton', 'Change default S3 region')
+            )
+            .then((selection: string | undefined) => {
+                if (selection === 'Change default S3 region') {
+                    vscode.commands.executeCommand('workbench.action.openSettings', 'aws.s3.defaultRegion')
```

Comment:
Should I leave a TODO on every place it was used? Or can I document that it is to consider in the future?

Reply:
I wouldn't leave a bunch of notes, but if you can find a place that makes sense you can add a note describing the issues with using a constant region. Generally I wouldn't leave in TODOs describing anything super specific since it's extremely likely the implementation will change over time, so the comments will just be noise.